feat: Added Params for CA Certs in MSSQL Config

pangjunrong opened the pull request:
When establishing an SSL connection to MSSQL currently, bb8 times out due to a mismatch in CA certificates since underlying default behavior from _tiberius_ is to validate server certificate against system-truststore. 

This change enables 2 new parameters in the connection string for MSSQL to address custom CA certificate use cases ([description extracted from _tiberius_ config documentation](https://docs.rs/tiberius/latest/tiberius/struct.Config.html)):

- **trust_server_certificate**: If set, the server certificate will not be validated and it is accepted as-is
- **trust_server_certificate_ca**: If set, the server certificate will be validated against the given CA certificate in in addition to the system-truststore. Useful when using self-signed certificates on the server without having to disable the trust-chain.

This will resolve issue #509 and is a requested feature at the end of the thread.

wangxiaoying commented:
Hi @pangjunrong , thanks for the PR. It looks good to me.

Can you fix the [formatting problem](https://github.com/sfu-db/connector-x/actions/runs/11383850412/job/31711777889?pr=698) of the code? You can do it by running `cargo fmt`.

Also, can you add this new parameter to the documentation [here](https://sfu-db.github.io/connector-x/databases/mssql.html) as well? You can do this by editing [this markdown file](https://github.com/sfu-db/connector-x/blob/main/docs/databases/mssql.md).

pangjunrong commented:
Hey @wangxiaoying, thanks for the quick reply! I've fixed the formatting and added the new parameters into the documentation.

So I updated tiberius & bb8-tiberius to 0.7.3 and 0.8.0 respectively as those versions contain the first instances of _ca function but I noticed that anything above tiberius 0.5.16 no longer contained the QueryResult method (unfortunately deprecated) which is used across the MSSQL implementation. I've tried my hand at replacing it with the new QueryStream method; hoping you can help out on this if you see anything amiss!

wangxiaoying commented:
Thank you @pangjunrong !
